lib/theme/ColorInitializer.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:flutter/material.dart';
+
+enum ColorInitializer {
+  primary,
+  primaryContainer,
+  onPrimary,
+  secondary,
+  secondaryContainer,
+  onSecondary,
+  background,
+  onBackground,
+  surface,
+  onSurface,
+  onError
+}
+
+validateForeground(bgColor, context) {
+  switch (bgColor) {
+    case ColorInitializer.primary:
+    case ColorInitializer.primaryContainer:
+    case ColorInitializer.secondary:
+    case ColorInitializer.secondaryContainer:
+    case ColorInitializer.surface:
+    case ColorInitializer.background:
+      return getForegroundColor(bgColor, context);
+    default:
+      return throw Exception("Invalid Theme Color");
+  }
+}
+
+validateBackground(bgColor, context) {
+  switch (bgColor) {
+    case ColorInitializer.primary:
+    case ColorInitializer.primaryContainer:
+    case ColorInitializer.secondary:
+    case ColorInitializer.secondaryContainer:
+    case ColorInitializer.surface:
+    case ColorInitializer.background:
+      return getBackgroundColor(bgColor, context);
+    default:
+      return throw Exception("Invalid Theme Color");
+  }
+}
+
+getForegroundColor(ColorInitializer bgColor, context) {
+  switch (bgColor) {
+    case ColorInitializer.primary:
+    case ColorInitializer.primaryContainer:
+      return Theme.of(context).colorScheme.onPrimary;
+    case ColorInitializer.secondary:
+    case ColorInitializer.secondaryContainer:
+      return Theme.of(context).colorScheme.onSecondary;
+    case ColorInitializer.surface:
+      return Theme.of(context).colorScheme.onSurface;
+    case ColorInitializer.background:
+      return Theme.of(context).colorScheme.onBackground;
+    default:
+      return Theme.of(context).colorScheme.onBackground;
+  }
```

Comment:
replace `getForegroundColor` method body with the following code snippet.
```suggestion
  final colorScheme = Theme.of(context).colorScheme;

  final fgColorMap = {
    ColorInitializer.primary: colorScheme.onPrimary,
    ColorInitializer.primaryContainer: colorScheme.onPrimary,
    ColorInitializer.secondary: colorScheme.onSecondary,
    ColorInitializer.secondaryContainer: colorScheme.onSecondary,
    ColorInitializer.surface: colorScheme.onSurface,
    ColorInitializer.background: colorScheme.onBackground,
  };

  return fgColorMap[bgColor] ??
      (throw ArgumentError(
          'bgColor is not a valid background color checkout ColorInitializer enum class.'));
```

Reply:
Done